### PR TITLE
Update GenerateLayout.targets for Microsoft.CodeAnalysis.NetAnalyzers packaging structure changes 

### PR DESCRIPTION
**Pull request by mavasani**
React to package structure updates in https://github.com/dotnet/roslyn-analyzers/pull/6325 and https://github.com/dotnet/roslyn-analyzers/pull/6427 to copy props/targets/config files from the `buildtransitive` folder in the NuGet package

### CODE REVIEW

**Comment by mavasani** [comment]
@Youssef1313

**Comment by Youssef1313** [comment]
I think this should be synchronized with NetAnalyzers version update that has the change

**Comment by mavasani** [comment]
> I think this should be synchronized with NetAnalyzers version update that has the change

Yes, will have to wait for https://github.com/dotnet/roslyn-analyzers/pull/6427 to be merged and then get the new NetAnalyzers version update which has this fix.

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <ProductDependencies>
     <Dependency Name="Microsoft.TemplateEngine.Abstractions" Version="8.0.100-alpha.1.23061.3">
       <Uri>https://github.com/dotnet/templating</Uri>
-      <Sha>bf9cf318137c92fa79278dfa67b33dbca870f40a</Sha>
+      <Sha>bf9cf318137c92fa79278dfa67b33dbca870f40a</Sha>RR
```
Comment (mavasani) [inline eng/Version.Details.xml:6]: Fix typo
```suggestion
      <Sha>bf9cf318137c92fa79278dfa67b33dbca870f40a</Sha>
```

**Comment by mavasani** [comment]
@nagilson for review. The build is passing fine for all but Ubuntu. I am not sure if this is a CI issue, because the file that is being complained as missing is indeed present in the package.

**Comment by mavasani** [comment]
Closing in favor of https://github.com/dotnet/sdk/pull/29911

**Pull request closed without merging**